File: Interview Problems/Parking Lot/Java Code/domain/Receipt.java
package domain;

import java.time.LocalDateTime;
import java.util.UUID;

public class Receipt {

    public enum PaymentStatus {
        PENDING, SUCCESS, FAILED
    }

    private final UUID id;
    private final UUID ticketId;
    private final double totalFee;
    private final LocalDateTime exitTime;
    private PaymentStatus paymentStatus;

    public Receipt(UUID ticketId, double totalFee) {
        this.id = UUID.randomUUID();
        this.ticketId = ticketId;
        this.totalFee = totalFee;
        this.exitTime = LocalDateTime.now();
        this.paymentStatus = PaymentStatus.PENDING;
    }

    public void markAsPaid() {
        this.paymentStatus = PaymentStatus.SUCCESS;
    }

    public void markAsFailed() {
        this.paymentStatus = PaymentStatus.FAILED;
    }

    public UUID getId() { return id; }
    public UUID getTicketId() { return ticketId; }
    public double getTotalFee() { return totalFee; }
    public LocalDateTime getExitTime() { return exitTime; }
    public PaymentStatus getPaymentStatus() { return paymentStatus; }

    @Override
    public String toString() {
        return "Receipt{id=" + id + ", ticketId=" + ticketId
                + ", fee=" + totalFee + ", status=" + paymentStatus + "}";
    }
}

File: Interview Problems/Parking Lot/Python Code/domain/receipt.py
import uuid
from datetime import datetime
from domain.payment import PaymentStatus

class Receipt:
    def __init__(self, ticket_id: uuid.UUID, total_fee: float):
        self.id = uuid.uuid4()
        self.ticket_id = ticket_id
        self.total_fee = total_fee
        self.exit_time = datetime.now()
        self.payment_status = PaymentStatus.PENDING

    def mark_as_paid(self):
        self.payment_status = PaymentStatus.SUCCESS

    def mark_as_failed(self):
        self.payment_status = PaymentStatus.FAILED

    def __repr__(self):
        return f"Receipt(id={self.id}, ticket_id={self.ticket_id}, fee={self.total_fee}, status={self.payment_status.name})"
